napari/_qt/qt_viewer.py

```diff
@@ -248,6 +249,25 @@ def _open_folder(self):
         if folder not in {'', None}:
             self._add_files([folder])
 
+    def _add_path(self, path):
```

Comment:
I feel like this should go on the ViewerModel as it is Qt independent - probably even in the `AddLayersMixin` - thoughts?

Reply:
Can think about where to put `self._last_visited_dir = os.path.dirname(path)`, could pull that out and leave in QtViewer, possibly in one of the caller functions

Reply:
I agree, it's a good opportunity to rethink where some of this stuff lives.  I intended to revisit that once our vision for the general plugin architecture settled a bit (which i think is now! :)
I guess since this method only uses `self.viewer` but never uses `self`, it should be on the `Viewer`.  I like the mixin idea too... since, hey, it's adding something to the viewer!